tests/unit/utils/test_juju_utils.py

```diff
@@ -459,4 +460,26 @@ async def test_coumodel_wait_for_idle(mock_get_supported_apps, mocked_model):
     mocked_model.wait_for_idle.assert_awaited_once_with(
         apps=["app1", "app2"], timeout=timeout, idle_period=juju_utils.DEFAULT_MODEL_IDLE_PERIOD
     )
-    mock_get_supported_apps.assert_awaited_once_with()
+
+
+@pytest.mark.asyncio
+@patch("cou.utils.juju_utils.COUModel._get_supported_apps")
+@pytest.mark.parametrize("apps", [None, ["app1", "app2"]])
+async def test_coumodel_wait_for_idle_timeout(mock_get_supported_apps, apps, mocked_model):
+    """Test COUModel wait for model to be idle reach timeout."""
+    timeout = 60
+    mocked_model.wait_for_idle.side_effect = asyncio.exceptions.TimeoutError
+    model = juju_utils.COUModel(None)
+
+    with pytest.raises(WaitForApplications):
+        await model.wait_for_idle(timeout, apps=apps)
+
+    if apps is None:
+        mock_get_supported_apps.assert_awaited_once_with()
+        exp_apps = mock_get_supported_apps.return_value
+    else:
+        exp_apps = apps
```

Comment:
Non-blocking: Since in https://github.com/canonical/charmed-openstack-upgrader/pull/109 you separated test cases to avoid `if` statements in unit tests, I would prefer we do the same here and keep it consistent